Plutarch/Either.hs

```diff
@@ -11,3 +21,10 @@ data PEither (a :: PType) (b :: PType) (s :: S)
   | PRight (Term s b)
   deriving stock (GHC.Generic)
   deriving anyclass (Generic, HasDatatypeInfo, PlutusType, PEq, PShow)
+
+-- | 'PEither' destructor equivalent to 'either'
+peither :: Term (s :: S) ((a :--> c) :--> (b :--> c) :--> PEither a b :--> c)
```

Comment:
This is likely not efficient.

Reply:
Relative to what? It's slightly less efficient than `pmatch`:
```
peither {"exBudgetCPU":119192,"exBudgetMemory":500,"scriptSizeBytes":23}
pmatch {"exBudgetCPU":119192,"exBudgetMemory":500,"scriptSizeBytes":18}
```

where

```
    "peither" @| peither # psucc # psucc @-> \p -> passert (p # pcon (PLeft 42) #== 43)
    "pmatch"
      @| plam (\p-> pmatch p $ \case {PLeft n -> psucc # n; PRight n -> psucc # n})
      @-> \p -> passert (p # pcon (PLeft 42) #== 43)
```

Reply:
It's still used by `AssocMap.mapEitherWithKeyData` but this is not used any more, so it can all be removed I suppose.

Reply:
Plutarch 2.0 todo as with `pmaybe`